chainermn/communicators/pure_nccl_communicator.py

```diff
@@ -196,160 +166,16 @@ def _multi_node_mean_nccl(self, sendbuf, recvbuf,
-                j_min = j + 1;
-                continue;
-            }
-            break;
-        }
-        assert(tid >= params_size_csum[j]);
-        assert(tid < params_size_csum[j+1]);
-        int src_dtype = params_dtype[j];
-        int src_idx = tid - params_size_csum[j];
-        if (dst_dtype == NCCL_FLOAT16) {
-            half* dst = (half*) dst0;
-            if (src_dtype == NCCL_FLOAT16) {
-                dst[tid] = (half) (((half*) (params_dptr[j]))[src_idx]);
-            }
-            else if (src_dtype == NCCL_FLOAT32) {
-                dst[tid] = (half) (((float*) (params_dptr[j]))[src_idx]);
-            }
-        }
-        else if (dst_dtype == NCCL_FLOAT32) {
-            float* dst = (float*) dst0;
-            if (src_dtype == NCCL_FLOAT16) {
-                dst[tid] = (float) (((half*) (params_dptr[j]))[src_idx]);
-            }
-            else if (src_dtype == NCCL_FLOAT32) {
-                dst[tid] = (float) (((float*) (params_dptr[j]))[src_idx]);
-            }
-       }
-    }
-    ''', 'cupy_batched_pack_params')
-
-
-def _cupy_batched_unpack_params():
-    return chainer.cuda.raw(r'''
-#include <cupy/carray.cuh>
-#define NCCL_FLOAT16  6
-#define NCCL_FLOAT32  7
-    extern "C" __global__
-    void cupy_batched_unpack_params(
-            void *src0, int src_dtype, int n_elems,
-            unsigned long *params_dptr, int *params_dtype,
-            int *params_size_csum, int n_params) {
-        int tid = threadIdx.x + blockIdx.x * blockDim.x;
-        if (tid >= n_elems) return;
-        int j_min = 0;
-        int j_max = n_params - 1;
-        int j;
-        while (1) {
-            j = (j_min + j_max) / 2;
-            if (tid < params_size_csum[j]) {
-                j_max = j - 1;
-                continue;
-            }
-            if (tid >= params_size_csum[j+1]){
-                j_min = j + 1;
-                continue;
-            }
-            break;
-        }
-        assert(tid >= params_size_csum[j]);
-        assert(tid < params_size_csum[j+1]);
-        int dst_dtype = params_dtype[j];
-        int dst_idx = tid - params_size_csum[j];
-        if (src_dtype == NCCL_FLOAT16) {
-            half* src = (half*) src0;
-            if (dst_dtype == NCCL_FLOAT16) {
-                ((half*) (params_dptr[j]))[dst_idx] = (half) src[tid];
-            }
-            else if (dst_dtype == NCCL_FLOAT32) {
-                ((float*) (params_dptr[j]))[dst_idx] = (float) src[tid];
-            }
-        }
-        else if (src_dtype == NCCL_FLOAT32) {
-            float* src = (float*) src0;
-            if (dst_dtype == NCCL_FLOAT16) {
-                ((half*) (params_dptr[j]))[dst_idx] = (half) src[tid];
-            }
-            else if (dst_dtype == NCCL_FLOAT32) {
-                ((float*) (params_dptr[j]))[dst_idx] = (float) src[tid];
-            }
-       }
-    }''', 'cupy_batched_unpack_params')
+def _get_converting_kernel(src_dtype, dst_dtype, kernel_name):
```

Comment:
Could you remove `_get_converting_kernel()`, `_get_param_data_dtype()`, and `_get_param_grad_dtype()` ?
These functions are not used now.
These functions were deleted in other PR